tests/Integration/Behaviour/Features/Context/Domain/ImageSettings/ImageTypeContext.php

```diff
@@ -0,0 +1,216 @@
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain\ImageSettings;
+
+use Behat\Gherkin\Node\TableNode;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Command\AddImageTypeCommand;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Command\BulkDeleteImageTypeCommand;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Command\DeleteImageTypeCommand;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Command\EditImageTypeCommand;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Exception\ImageTypeNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Query\GetImageTypeForEditing;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\QueryResult\EditableImageType;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\ValueObject\ImageTypeId;
+use Tests\Integration\Behaviour\Features\Context\Domain\AbstractDomainFeatureContext;
+use Tests\Integration\Behaviour\Features\Context\Util\PrimitiveUtils;
+
+class ImageTypeContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @When I create an image type :imageTypeName with following properties:
+     */
+    public function createImageTypeUsingCommand(string $imageTypeName, TableNode $table)
+    {
+        $data = $this->fixDataType($table->getRowsHash());
+
+        $command = new AddImageTypeCommand(
+            $imageTypeName,
+            $data['width'],
+            $data['height'],
+            $data['products'],
+            $data['categories'],
+            $data['manufacturers'],
+            $data['suppliers'],
+            $data['stores']
+        );
+
+        /** @var ImageTypeId $imageTypeId */
+        $imageTypeId = $this->getCommandBus()->handle($command);
+
+        $this->getSharedStorage()->set($imageTypeName, $imageTypeId->getValue());
+    }
+
+    /**
+     * @When I edit image type :imageTypeName with following properties:
+     */
+    public function editImageTypeUsingCommand(string $imageTypeName, TableNode $table)
+    {
+        $data = $this->fixDataType($table->getRowsHash());
+        $this->getSharedStorage()->exists($imageTypeName);
+
+        $command = new EditImageTypeCommand($this->getSharedStorage()->get($imageTypeName));
+        $command->setName($imageTypeName);
```

Comment:
It would be nice if this step could test partial updates as well (meaning you need to check if the table node contains the field, only set it when present and add some scenarios in the behat tests that only update partial elements and check that the others didn't change), especially since you already handled the partial assertion This would also validate that Doctrine partial update works properly (maybe not that the SQL query is optimized, but at least that it doesn't erase unwanted values)

But I won't block this huge PR for this, let's hope it will work fine when we try and set up a PATCH API on this 😅